components/tag-list/tag-list.js

```diff
@@ -0,0 +1,96 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ArrowKeysMixin } from '../../mixins/arrow-keys-mixin.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+
+const ITEM_SPACING = 6;
+
+class TagList extends ArrowKeysMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: A description of the tag list for additional accessibility context
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * @ignore
+			 */
+			role: { type: String, reflect: true },
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.tag-list-container {
+				display: flex;
+				flex-wrap: wrap;
+				margin: -${ITEM_SPACING}px -${ITEM_SPACING}px 0 0;
+				padding: 0;
+				position: relative;
+			}
+			::slotted(*) {
+				margin: ${ITEM_SPACING}px ${ITEM_SPACING}px 0 0;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		/** @ignore */
+		this.role = 'application';
+		/** @ignore */
+		this.arrowKeysDirection = 'leftrightupdown';
+		this._items = [];
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		requestAnimationFrame(() => {
+			this._items = this._getTagListItems();
+			if (this._items.length > 0) this._items[0].setAttribute('tabindex', 0);
```

Comment:
Not that you necessarily need to support it, but technically the first tag could be `hidden`, in such case focus could never get into the tags list and arrow keys wouldn't work.  Just mentioning in case this was something that might be expected.

Reply:
For sure, I'll definitely need to handle hidden tags anyways once we start truncating after some number of lines.  Haven't decided yet if that will be a separate `_visibleItems` array that I could use here instead, or if I want a `truncatingIndex` and I'll need to check whether anything before that is also hidden.